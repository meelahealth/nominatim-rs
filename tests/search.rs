#[tokio::test]
async fn search() {
    let client = nominatim::Client::new(
        url::Url::parse("https://nominatim.openstreetmap.org/").unwrap(),
        "nominatim-rust/0.1.0 test-suite".to_string(),
        Some("john_t@mailo.com".to_string()),
    )
    .unwrap();

    let search = client
        .search(
            nominatim::search::SearchQueryBuilder::default()
                .address_details(true)
                .location_query(nominatim::search::LocationQuery::Generalised {
                    q: "bakery in berlin wedding".to_string(),
                })
                .limit(Some(1))
                .build()
                .unwrap(),
        )
        .await
        .unwrap();

    let search_expected: nominatim::types::Response = serde_json::from_str(include_str!("./search_expected.json")).unwrap();

    println!("{}", serde_json::to_string_pretty(&search).unwrap());

    assert_eq!(search, vec![search_expected]);
}
