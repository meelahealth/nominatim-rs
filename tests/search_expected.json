{
    "place_id": 1346017,
    "osm_type": "node",
    "osm_id": 317179427,
    "lon": "13.35918",
    "lat": "52.5460941",
    "licence": "Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright",
    "address": {
        "continent": null,
        "country": "Deutschland",
        "country_code": "de",
        "state": "Berlin",
        "city": "Berlin",
        "borough": "Mitte",
        "suburb": "Wedding",
        "neighbourhood": "Sprengelkiez",
        "road": "Müllerstraße",
        "shop": "Kamps",
        "postcode": "13353"
    },
    "display_name": "Kamps, Müllerstraße, Sprengelkiez, Wedding, Mitte, Berlin, 13353, Deutschland",
    "icon": "https://nominatim.openstreetmap.org/ui/mapicons//shopping_bakery.p.20.png",
    "class": "shop",
    "type": "bakery",
    "importance": 0.30100000000000005,
    "boundingbox": [
        "52.5460441",
        "52.5461441",
        "13.35913",
        "13.35923"
    ]
}
